### ISSUE TITLE
Nan interpolation for descending direction

### ISSUE THREAD

**Issue by Puumanamana** [sanity/pav.rs#1]
Hi,

Thanks for developing this crate, it's very convenient for my application! 
I was able to make it work pretty easily. However, I'm having issues with the interpolate function in the case of a descending direction. Here's an example:

```rust
let regression = IsotonicRegression::new_descending(
        &[
            Point::new(0.0, 3.0),
            Point::new(1.0, 2.0),
            Point::new(2.0, 1.0)
        ]
);
println!("{:}", regression.interpolate(0.5));
// Returns NaN
``` 
This issue however doesn't occur on an ascending direction when I reverse the order of the y coordinates. Any idea what's happening?

Thank you!

**Comment by sanity** [comment]
Thanks for the bug report. This was due to an error in the logic for descending. I've replaced this with a more straightforward implementation and added a test for this problem, released as 0.2.4. Please let me know if you have further issues.

**Comment by Puumanamana** [comment]
It worked, thanks for the fix!
